crates/sui-core/src/consensus_adapter.rs

```diff
@@ -194,7 +194,34 @@ pub trait SubmitToConsensus: Sync + Send + 'static {
         &self,
         transactions: &[ConsensusTransaction],
         epoch_store: &Arc<AuthorityPerEpochStore>,
-    ) -> SuiResult;
+    ) -> SuiResult<SubmitResponse>;
+}
+
+pub enum BlockStatus {
+    Sequenced,
+    GarbageCollected,
+}
```

Comment:
Not _that_ necessary doing this, I could have just used the Mysticeti enums, but since we are still having Narwhal around and we keep things a bit more generic I've went for having a SUI API for this. Can refactor if we consider it too much.